L1Trigger/L1THGCal/plugins/backend/HGCalStage1TruncationWrapper.cc

```diff
@@ -0,0 +1,110 @@
+
+  void convertAlgorithmOutputs(const l1thgcfirmware::HGCalTriggerCellSACollection& fpga_tcs_out,
+                               const std::vector<edm::Ptr<l1t::HGCalTriggerCell>>& fpga_tcs_original,
+                               std::vector<edm::Ptr<l1t::HGCalTriggerCell>>& fpga_tcs_trunc) const;
+
+  void eventSetup(const edm::EventSetup& es) { triggerTools_.eventSetup(es); }
+
+  HGCalTriggerTools triggerTools_;
+  HGCalStage1TruncationImplSA theAlgo_;
+  l1thgcfirmware::Stage1TruncationConfig theConfiguration_;
+};
+
+HGCalStage1TruncationWrapper::HGCalStage1TruncationWrapper(const edm::ParameterSet& conf)
+    : HGCalStage1TruncationWrapperBase(conf),
+      theAlgo_(),
+      theConfiguration_(conf.getParameter<bool>("doTruncation"),
+                        conf.getParameter<double>("rozMin"),
+                        conf.getParameter<double>("rozMax"),
+                        conf.getParameter<unsigned>("rozBins"),
+                        conf.getParameter<std::vector<unsigned>>("maxTcsPerBin"),
+                        conf.getParameter<std::vector<double>>("phiSectorEdges")) {}
+
+void HGCalStage1TruncationWrapper::convertCMSSWInputs(const std::vector<edm::Ptr<l1t::HGCalTriggerCell>>& fpga_tcs,
+                                                      l1thgcfirmware::HGCalTriggerCellSACollection& fpga_tcs_SA) const {
+  fpga_tcs_SA.clear();
+  fpga_tcs_SA.reserve(fpga_tcs.size());
+  unsigned int itc = 0;
+  for (auto& tc : fpga_tcs) {
+    fpga_tcs_SA.emplace_back(tc->position().x(),
+                             tc->position().y(),
+                             tc->position().z(),
+                             triggerTools_.zside(tc->detId()),
+                             triggerTools_.layerWithOffset(tc->detId()),
+                             tc->eta(),
+                             tc->phi(),
+                             tc->pt(),
+                             tc->mipPt(),
+                             itc);
+    ++itc;
+  }
+}
+
+void HGCalStage1TruncationWrapper::convertAlgorithmOutputs(
+    const l1thgcfirmware::HGCalTriggerCellSACollection& fpga_tcs_out,
+    const std::vector<edm::Ptr<l1t::HGCalTriggerCell>>& fpga_tcs_original,
+    std::vector<edm::Ptr<l1t::HGCalTriggerCell>>& fpga_tcs_trunc) const {
+  for (unsigned itc = 0; itc < fpga_tcs_out.size(); ++itc) {
+    unsigned tc_cmssw_id = fpga_tcs_out[itc].index_cmssw();
+    fpga_tcs_trunc.push_back(fpga_tcs_original[tc_cmssw_id]);
+  }
+}
+
+void HGCalStage1TruncationWrapper::process(const std::vector<edm::Ptr<l1t::HGCalTriggerCell>>& fpga_tcs,
+                                           std::vector<edm::Ptr<l1t::HGCalTriggerCell>>& tcs_out) const {
+  l1thgcfirmware::HGCalTriggerCellSACollection fpga_tcs_SA;
+  convertCMSSWInputs(fpga_tcs, fpga_tcs_SA);
+
+  l1thgcfirmware::HGCalTriggerCellSACollection tcs_out_SA;
+  unsigned ec = theAlgo_.run(fpga_tcs_SA, theConfiguration_, tcs_out_SA);
```

Comment:
Would be easier to understand what `ec` is if it is named `error` or similar.

Reply:
Now called `error_code`